grants_tagger/models.py

```diff
@@ -0,0 +1,604 @@
+                Y_pred_i = classifier.predict_proba(X_vec)
+            elif self.threshold != 0.5:
+                Y_pred_i = classifier.predict_proba(X_vec) > self.threshold
+                Y_pred_i = sp.csr_matrix(Y_pred_i)
+            else:
+                Y_pred_i = classifier.predict(X_vec)
+            Y_pred.append(Y_pred_i)
+
+        if return_probabilities:
+            Y_pred = np.hstack(Y_pred)
+        else:
+            Y_pred = sp.hstack(Y_pred)
+        return Y_pred
+
+    def predict(self, X):
+        return self._predict(X)
+
+    def predict_proba(self, X):
+        return self._predict(X, return_probabilities=True)
+
+    def save(self, model_path):
+        if model_path != self.model_path:
+            print(f"{model_path} is different from self.model_path {self.model_path}. This will result in model and meta.json be saved in different paths")
+
+        meta = {
+            "name": "MeshTfidfSVM",
+            "approach": "mesh-tfidf-svm",
+            "y_batch_size": self.y_batch_size,
+            "nb_labels": self.nb_labels
+        }
+        meta_path = os.path.join(model_path, "meta.json")
+        with open(meta_path, "w") as f:
+            f.write(json.dumps(meta))
+
+    def load(self, model_path):
+        vectorizer_path = os.path.join(model_path, "vectorizer.pkl")
+        with open(vectorizer_path, "rb") as f:
+            self.vectorizer = pickle.loads(f.read())
+
+        meta_path = os.path.join(model_path, "meta.json")
+        with open(meta_path, "r") as f:
+            meta = json.loads(f.read())
+        self.set_params(**meta)
+
+        self.model_path = model_path
+
+def create_model(approach, parameters=None):
+    if approach == 'tfidf-svm':
+        model = Pipeline([
+            ('tfidf', TfidfVectorizer(
+                stop_words='english', max_df=0.95,
+                min_df=0.0, ngram_range=(1,1)
+            )),
+            ('svm', OneVsRestClassifier(SVC(kernel='linear', probability=True)))
+        ])
+    elif approach == 'bert-svm':
+        model = Pipeline([
+            ('bert', BertVectorizer(pretrained='bert')),
+            ('svm', OneVsRestClassifier(SVC(kernel='linear', probability=True)))
+        ])
+    elif approach == 'scibert-svm':
+        model = Pipeline([
+            ('scibert', BertVectorizer(pretrained='scibert')),
+            ('svm', OneVsRestClassifier(SVC(kernel='linear', probability=True)))
+        ])
+    elif approach == 'spacy-textclassifier':
+        model = SpacyClassifier()
+    elif approach == 'bert':
+        model = BertClassifier()
+    elif approach == 'scibert':
+        model = BertClassifier(pretrained='scibert')
+    elif approach == 'classifierchain-tfidf-svm':
+        model = Pipeline([
+            ('tfidf', TfidfVectorizer(
+                stop_words='english', max_df=0.95,
+                min_df=0.0, ngram_range=(1,1)
+            )),
+            ('svm', ClassifierChain(classifier=SVC(kernel='linear', probability=True)))
```

Comment:
These parameters for SVC - did you get it after doing some sort of grid-search? Not related to this PR specifically, but wondering whether changing `C` and kernel to `RBF` will give you a boost

Reply:
I think RBF does not have probability or you need to do something else to get a probability. These params are just init values, params are also set in the init files for every experiment. So possibly it would be cleaner not to set any param and to always override all params using init files. I will resolve that when I move to DVC